source/PlayerInfo.cpp

```diff
@@ -1356,7 +1367,21 @@ void PlayerInfo::SetShipOrder(const vector<shared_ptr<Ship>> &newOrder)
 	// Check if the incoming vector contains the same elements
 	if(std::is_permutation(ships.begin(), ships.end(), newOrder.begin()))
 	{
+		shared_ptr<Ship> oldFirstShip = ships.front();
 		ships = newOrder;
+		shared_ptr<Ship> newFirstShip = ships.front();
+		// Check if the position of the flagship has changed.
+		if(newFirstShip != oldFirstShip && Preferences::Has("Automatically unpark flagship"))
+		{
+			// Find the position of the previous flagship in the new vector.
+			vector<shared_ptr<Ship>>::iterator pos = find(ships.begin(), ships.end(), oldFirstShip);
+			// Check if the ship in the first position can be a flagship and is in the current system.
+			if(newFirstShip->CanBeFlagship() && newFirstShip->GetSystem() == system)
+			{
+				newFirstShip->SetIsParked(false);
+				(*pos)->SetIsParked(true);
+			}
+		}
```

Comment:
```suggestion
		shared_ptr<Ship> oldFirstShip = ships.front();
		ships = newOrder;
		shared_ptr<Ship> newFirstShip = ships.front();
		// Check if the position of the flagship has changed and
		// if the ship in the first position can be a flagship and is in the current system.
		if(newFirstShip != oldFirstShip && Preferences::Has("Automatically unpark flagship")
			&& newFirstShip->CanBeFlagship() && newFirstShip->GetSystem() == system)
		{
			newFirstShip->SetIsParked(false);
			oldFirstShip->SetIsParked(true);
		}
```

Reply:
Shouldnt I change the shared_ptr to iterator aswell?